Cubical/Foundations/Isomorphism.agda

```diff
@@ -0,0 +1,121 @@
+{-
+
+Theory about equivalences (definitions are in Core/Glue.agda)
+
+- isEquiv is a proposition ([isPropIsEquiv])
+- Any isomorphism is an equivalence ([isoToEquiv])
+- transport is an equivalence ([transportEquiv])
+
+-}
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Foundations.Isomorphism where
+
+open import Cubical.Core.Everything
+open import Cubical.Foundations.HLevels
+
+-- Proof using isPropIsContr. This is slow and the direct proof below is better
+isPropIsEquiv' : ∀ {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} (f : A → B) → isProp (isEquiv f)
+equiv-proof (isPropIsEquiv' f u0 u1 i) y =
+  isPropIsContr (u0 .equiv-proof y) (u1 .equiv-proof y) i
+
+-- Direct proof that computes quite ok (can be optimized further if
+-- necessary, see:
+-- https://github.com/mortberg/cubicaltt/blob/pi4s3_dimclosures/examples/brunerie2.ctt#L562
+isPropIsEquiv : ∀ {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} (f : A → B) → isProp (isEquiv f)
+equiv-proof (isPropIsEquiv f p q i) y =
+  let p2 = p .equiv-proof y .snd
+      q2 = q .equiv-proof y .snd
+  in p2 (q .equiv-proof y .fst) i
+   , λ w j → hcomp (λ k → λ { (i = i0) → p2 w j
+                            ; (i = i1) → q2 w (j ∨ ~ k)
+                            ; (j = i0) → p2 (q2 w (~ k)) i
+                            ; (j = i1) → w })
+                   (p2 w (i ∨ j))
+
+equivEq : ∀ {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} (e f : A ≃ B) → (h : e .fst ≡ f .fst) → e ≡ f
+equivEq e f h = λ i → (h i) , isProp→PathP isPropIsEquiv h (e .snd) (f .snd) i
+
+
+-- Section and retract
+module _ {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} where
+  section : (f : A → B) → (g : B → A) → Set ℓ'
+  section f g = ∀ b → f (g b) ≡ b
+
+  -- NB: `g` is the retraction!
+  retract : (f : A → B) → (g : B → A) → Set ℓ
+  retract f g = ∀ a → g (f a) ≡ a
+
+-- Any iso is an equivalence
```

Comment:
Can you please package the data of an isomorphism into a Record?  (Maybe with the name `Iso`)

Reply:
Maybe "isIso f" and "Iso A B" analogous to "isEquiv f" and "A \simeq B"?

Reply:
I don't like the term `isIso` as it is not a proposition... I really like the convention that everything that starts with `is` is a proposition (which goes back to Voevodsky). 

How about `isoStruct` or something to emphasize that it is a structure?

Reply:
Yes. Are you ok with calling the witnesses for "section f inverse" "rightInv" and similar for "rectract..."?

Reply:
Should I keep the types of "isoToEquiv" and "isoToPath" or should I use the record everywhere?

Reply:
Use the record everywhere

Reply:
> Yes. Are you ok with calling the witnesses for "section f inverse" "rightInv" and similar for "rectract..."?

Sounds good to me